File: ios/Plugin/CapReadNativeSettingsPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapReadNativeSettingsPlugin, "CapReadNativeSettings",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
)

File: ios/Plugin/CapReadNativeSettingsPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapReadNativeSettingsPlugin)
public class CapReadNativeSettingsPlugin: CAPPlugin {
    private let implementation = CapReadNativeSettings()

    @objc func read(_ call: CAPPluginCall) {
        let key = call.getString("key") ?? ""
        let value = Bundle.main.object(forInfoDictionaryKey: key) as? String
        call.resolve([
            "value": value as Any
        ])
    }
}
